File: pinyougou/pinyougou-manage-web/src/main/webapp/js/controller/brandController.js
app.controller("brandController", function ($scope, $controller, brandService) {

    //继承baseController
    $controller("baseController",{$scope: $scope})

    // 查询所有列表数据并绑定到list对象
    $scope.findAll = function () {
        brandService.findAll().success(function (response) {
            $scope.reloadList();
        });
    };

    //分页查询
    // $scope.findPage = function (page, rows) {
    //     $http.get("../brand/findPage.do?page=" + page + "&rows=" + rows).success(function (response) {
    //         //更新记录列表 （点击哪页返回哪页数据）
    //         $scope.list = response.rows;
    //         //更新总页数
    //         $scope.paginationConf.totalItems = response.total;
    //     });
    // };

    //新建品牌
    $scope.save = function () {
        var method = "add";
        if ($scope.entity.id != null) {
            method = "update";
        }

        brandService.save(method, $scope.entity).success(function (response) {
            if (response.success) {
                $scope.reloadList();
                //提交后清除数据,或可在按钮点击时候处理 ng-click="entity={}
                //$scope.entity={};
            } else {
                alert(response.message);
            }
        });
    };

    //查找一个品牌数据
    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity = response;
        })
    };

    $scope.delete = function () {
        alert($scope.selectedIds)
        if ($scope.selectedIds.length < 1) {
            alert("请选择删除项!");
            return;
        }
        if (confirm("确定要进行删除吗?")) {
            brandService.delete($scope.selectedIds).success(function (response) {
                if (response.success) {
                    $scope.reloadList();

                    $scope.selectedIds = [];
                } else {
                    alert(response.message);
                }

            });
        }
    };

    $scope.searchEntity = {};
    $scope.search = function (page, rows) {
        brandService.search(page, rows, $scope.searchEntity)
            .success(function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems = response.total;
            });
    };
});

File: pinyougou/pinyougou-manage-web/src/main/java/com/pinyougou/manage/controller/BrandController.java
package com.pinyougou.manage.controller;


import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import com.pinyougou.vo.PageResult;
import com.pinyougou.vo.Result;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RequestMapping("/brand")
@RestController
public class BrandController {

    @Reference
    private BrandService brandService;

    @GetMapping("/testPage")
    public List<TbBrand> testPage(Integer page, Integer rows) {
//        return brandService.testPage(page, rows);
        return (List<TbBrand>) brandService.findPage(page, rows).getRows();
    }


    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
//        return brandService.queryAll();
        return brandService.findAll();
    }

    @GetMapping("/findPage")
    public PageResult findPage(
            @RequestParam(value = "page", defaultValue = "1") Integer page,
            @RequestParam(value = "rows", defaultValue = "5") Integer rows
    ) {
        return brandService.findPage(page, rows);
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand) {
        try {
            brandService.add(brand);
            return Result.ok("新增成功");
        } catch (Exception e) {
            e.printStackTrace();
            return Result.fail("新增失败");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(Long id) {
        return brandService.findOne(id);
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand) {

        try {
            brandService.update(brand);
            return Result.ok("修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return Result.fail("修改失败");
        }
    }

    @GetMapping("/delete")
    public Result delete(Long[] ids) {
        try {
            brandService.deleteByIds(ids);
            return Result.ok("删除成功!");
        } catch (Exception e) {
            e.printStackTrace();
        }
        return Result.fail("删除失败!");
    }

    @PostMapping("/search")
    public PageResult search(
            @RequestBody TbBrand tbBrand,
            @RequestParam(value = "page",defaultValue = "1") Integer page,
            @RequestParam(value = "rows",defaultValue = "5") Integer rows) {
        return brandService.search(tbBrand, page, rows);

    }

    /**
     * 查询品牌列表,返回数据格式符合select2格式
     * @return
     */
    @GetMapping("/selectOptionList")
    public List<Map<String, Object>> selectOptionList() {
        return brandService.selectOptionList();
    }

}
